src/mapbox/mapbox.js

```diff
@@ -211,11 +211,23 @@ export default class Mapbox {
   // (e.g. until "componentDidUpdate")
   resize() {
     this._map.resize();
+    return this;
+  }
+
+  // Force redraw the map now. Typically resize() and jumpTo() is reflected in the next
+  // render cycle, which is managed by Mapbox's animation loop.
+  // This removes the synchronization issue caused by requestAnimationFrame.
+  redraw() {
+    const map = this._map;
     // map render will throw error if style is not loaded
-    if (this._map.isStyleLoaded()) {
-      this._map._render();
+    if (map.isStyleLoaded()) {
+      map._render();
+      // cancel the scheduled update
```

Comment:
why cancel the scheduled update can force redraw?

Reply:
`_render` contains the redraw logic. We only cancel the scheduled update because all changes have been flushed.

Reply:
Maybe suggest to mapbox that they provide a public API for this? And add a link to that github issue in a comment here?

Also, do we gain much by canceling the next redraw? FPS when resizing/transition? 
If not, seems the less assumptions we make (about how mapbox manages dirty state and schedules draws etc), the better, and allowing an occasional extra draw might not be too bad?

Reply:
> do we gain much by canceling the next redraw?

As far as I can tell mapbox's `_render` method does not check the dirty state. Instead, they schedule a redraw when the dirty state changes.

Since we call this every time map prop updates, it can be fairly significant during transition and interaction.